index.js

```diff
@@ -105,12 +105,19 @@ var defaultCommands = {
       'Did you forget to wrap the key(s) in an array?',
       value
     );
+    var deletingKeys = [];
     var originalValue = nextObject;
     for (var i = 0; i < value.length; i++) {
       var key = value[i];
       if (Object.hasOwnProperty.call(originalValue, key)) {
-        originalValue = nextObject === object ? copy(object) : nextObject;
```

Comment:
What about doing something like this?

```js
    var originalValue = nextObject;
    for (var i = 0; i < value.length; i++) {
      var key = value[i];
      if (Object.hasOwnProperty.call(originalValue, key)) {
        if (nextObject === object) nextObject = copy(object);
        delete nextObject[key];
      }
    }
    return nextObject;
```

it should take care of the problem of recopying the object without having to reiterate over the keys